processing/src/main/java/org/apache/druid/segment/data/VSizeLongSerde.java

```diff
@@ -330,7 +329,7 @@ public void write(long value) throws IOException
         curByte = (byte) value;
         first = false;
       } else {
-        curByte = (byte) ((curByte << 4) | ((value >> (numBytes << 3)) & 0xF));
+        curByte = (byte) ((curByte << 4) | ((value >>> (numBytes << 3)) & 0xF));
```

Comment:
Was this a bug fix? If so: it's on the write side; does that mean there might be bad segments out there, or is there some reason that this line wouldn't have affected any already-written data that people might have? (Maybe negative numbers were never fed to this method.)

Reply:
oops, this was an accidental change, will revert